library/threading.c

```diff
@@ -29,6 +29,14 @@
 
 #include "mbedtls/threading.h"
 
+#if !defined(_WIN32) && (defined(__unix__) || \
+    (defined(__APPLE__) && defined(__MACH__)))
+#include <unistd.h>
+#if !defined(_POSIX_VERSION)
```

Comment:
I think it would be acceptable for now to assume that all POSIX platforms have `gmtime_r` and refine our test later if and when we get feedback from users that they are actually using a platform where this isn't the case.

This would already be a vast improvement over the current situation (assume all non-windows platforms have `gmtime_r`), and would fix the part of the problem that we know to be real.